pkg/gce-pd-csi-driver/node.go

```diff
@@ -327,6 +333,58 @@ func (ns *GCENodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStage
 	return &csi.NodeStageVolumeResponse{}, nil
 }
 
+func (ns *GCENodeServer) volumeCompatible(mountedPath, devicePath string, volumeCapability *csi.VolumeCapability) error {
+	if blk := volumeCapability.GetBlock(); blk != nil {
+		// If the volume capability request is type "Block" we don't care what
+		// format the disk is in because user could have re-formatted to a
+		// different type
+		// TODO: Need to check whether loopback device file is the same as the
+		// device path given, the following code to check for mount path doesn't
+		// work for block devices
+		return nil
+	}
+
+	// Part 1: Check that volume mounted at mountedPath is the same as one at
+	// devicePath
+	devicePathDev, err := filepath.EvalSymlinks(devicePath)
+	if err != nil {
+		return fmt.Errorf("failed to find backing disk for devicePath %s: %v", devicePath, err)
+	}
+	// df -P returns with the rows containing the backing disk and the location
+	// of the mount. awk proccesses the result by getting the line with the
+	// location of the mount we're looking for and printing out the backing
+	// disk. the resulting output should be a path to a device such as
+	// "/dev/sda"
+	mountedPathDevBytes, err := ns.Mounter.Exec.Command("sh", "-c", fmt.Sprintf("df -P | awk '$6==\"%s\"{print $1}'", mountedPath)).CombinedOutput()
```

Comment:
I'm not sure if df works for bind mounts. Mounter I believe has GetMountRefs() which I think handles bind mounts